File: backend_java/src/main/java/com/photohub/model/PaymentIssue.java
package com.photohub.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "payment_issues")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PaymentIssue {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long paymentIssueId;

    @Column(nullable = false)
    private Long paymentId;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false)
    private String issueType; // e.g., "CHARGEBACK", "REFUND_REQUEST"

    @Column(length = 1024)
    private String description;

    @Column(nullable = false)
    private String status = "OPEN"; // OPEN, RESOLVED, REJECTED

    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime resolvedAt;
    private Long resolvedByAdmin;
}

File: backend-dotnet/Models/PaymentIssue.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace PhotoHub.Analytics.Models
{
    public class PaymentIssue
    {
        [Key]
        [JsonPropertyName("issue_id")]
        public long IssueId { get; set; }

        [JsonPropertyName("payment_id")]
        public long PaymentId { get; set; }

        [JsonPropertyName("user_id")]
        public long UserId { get; set; }

        [JsonPropertyName("issue_type")]
        public string IssueType { get; set; } = string.Empty;

        [JsonPropertyName("description")]
        public string Description { get; set; } = string.Empty;

        [JsonPropertyName("status")]
        public string Status { get; set; } = "Open";

        [JsonPropertyName("resolved_by_admin")]
        public long? ResolvedByAdmin { get; set; }

        [JsonPropertyName("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
